tools/checked-c-convert/Utils.cpp

```diff
@@ -84,4 +93,82 @@ getFunctionDeclarationEnd(FunctionDecl *FD, SourceManager &S)
   } else {
     return FD->getSourceRange().getEnd();
   }
+}
+
+clang::CheckedPointerKind getCheckedPointerKind(InteropTypeExpr *itypeExpr) {
+  TypeSourceInfo * interopTypeInfo = itypeExpr->getTypeInfoAsWritten();
+  const clang::Type *innerType = interopTypeInfo->getType().getTypePtr();
```

Comment:
> You can just use itypeExpr->getType(), instead of getting the type as it is written. You can also use getAs to get the PointerType (if there is one), and then use T->getKind(), instead of doing a case by case analysis here.

Agree. Fixed it.